### PR TITLE
added contribution guidelines

### PR DESCRIPTION
**Pull request by SwartzCr**
@cooperq please read this over and give feedback !

### CODE REVIEW

File: doc/CONTRIBUTION-GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+### GIT ETTIQUITE
+
+When contributing to via git
+
+If You are an owner:
+
+ - Checkout the main repository and make a local branch for your change
+
+ - Make your edits locally and then commit them (preferably squashed into one commit)
+   to the origin repo as branch
```
Comment (cooperq) [inline doc/CONTRIBUTION-GUIDELINES.md]: I thought we decided to not squash things, am I misremembering this?

File: doc/CONTRIBUTION-GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ - Follow the same procedure but with a personal fork of the repository
+
+ - Ensure that your fork is up to date before committing your changes
+
+### GENERAL ETTIQUITE
+
+When making a code change:
+
+ - Make sure all your tests pass, and take time to write new tests for your
+   additions
```
Comment (cooperq) [inline doc/CONTRIBUTION-GUIDELINES.md]: Also it's nice if you write regression tests for a bug you are fixing, or if the part of code you are touching is not already covered by tests write some extra tests to cover it.

File: doc/CONTRIBUTION-GUIDELINES.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+ - Take time to ensure your code works as expected, effort skipped here merely
+   pushes the burden onto whomever reviews your commit
+
+ - Make sure commented lines and print statements are removed before submition
```
Comment (cooperq) [inline doc/CONTRIBUTION-GUIDELINES.md]: I would expand this to say run jshint on your code and fix any linting errors before you commit.

File: doc/CONTRIBUTION-GUIDELINES.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+ - If there are changes that effect the visible operations of Privacy Badger
+   that should be tested as part of code review, write up a short description of:
+   the issue your fixing, how to reproduce that issue, what the fix should look
+   like
+
+When reviewing a code change:
+
+ - Ensure all the tests pass, and take time to sanity check the changes in browser
```
Comment (cooperq) [inline doc/CONTRIBUTION-GUIDELINES.md]: I would also say to make sure that the code is consistent with our style guidelines and lint-free. This should be said to contributors as well.

**Comment by cooperq** [comment]
Other than my above comments this looks fine. The main points are that we should further emphasize the importance of good test and remind people to de-lint their code and follow our style guidelines.

**Comment by SwartzCr** [comment]
@cooperq I made your suggested edits, feel free to merge if you're content with them

**Comment by cooperq** [comment]
Looks great!

**Pull request merged**